client/src/pages/DashboardPage.jsx

```diff
@@ -44,6 +49,18 @@ const DashboardPage = () => {
           setisLoading(false);
         });
     }
+
+    const socket = socketIOClient(SOCKET_URL, {
+      query: `type=${sort}&term=${debouncedSearch}`,
+    });
```

Comment:
So does this create a new connection every time the query changes? If it doesn't, that is awesome and this works perfectly

If it does, maybe we should set it up so that we connect once, and then in this useEffect we should emit an event that the query changed and update the back-end with the new query details

Reply:
Maybe moving this connection up outside the component code and sub/unsub/disconnect in another useEffect with an empty deps array could do the trick to avoid another connection on search updates.

Another idea, if we'd want to make sure to keep the same socket connection for the whole user session, could be to initialize it earlier (app init or auth) and store it somewhere in the user context.

Reply:
😄 @splantebulldozer and I actually discussed about it yesterday. Yes it did.

Fixed now in commit 753c15718f28736608e6a17833fbc7ad4a0d8b0f

Can you please resolve conversation once validated?